runtime/near-vm-runner/src/prepare.rs

```diff
@@ -23,18 +21,48 @@ pub(crate) const WASM_FEATURES: wasmparser::WasmFeatures = wasmparser::WasmFeatu
     memory64: false,
 };
 
+/// Loads the given module given in `original_code`, performs some checks on it and
+/// does some preprocessing.
+///
+/// The checks are:
+///
+/// - module doesn't define an internal memory instance,
+/// - imported memory (if any) doesn't reserve more memory than permitted by the `config`,
+/// - all imported functions from the external environment matches defined by `env` module,
+/// - functions number does not exceed limit specified in VMConfig,
+///
+/// The preprocessing includes injecting code for gas metering and metering the height of stack.
+pub fn prepare_contract(original_code: &[u8], config: &VMConfig) -> Result<Vec<u8>, PrepareError> {
+    wasmparser::Validator::new()
+        .wasm_features(WASM_FEATURES)
+        .validate_all(original_code)
+        .map_err(|_| PrepareError::Deserialization)?;
+
+    match config.limit_config.stack_limiter_version {
+        // Support for old protocol versions, where we incorrectly didn't
+        // account for many locals of the same type.
+        //
+        // See `test_stack_instrumentation_protocol_upgrade` test.
```

Comment:
FWIW, there's a cool trick to make such links to tests checkable (ie, the test fails if it doesn't actually touch the relevant code): https://docs.rs/cov-mark/latest/cov_mark/.

I am still on the fence where maintaining such coverage marks helps overall.